mediaRouter.js

```diff
@@ -0,0 +1,90 @@
+const express = require("express");
+const { orderBy, slice } = require("lodash");
+const utils = require("./utils/common");
+const gitUtils = require("./git-utils");
+const media = require("./utils/media");
+
+module.exports = (dbDir, repositoryRoot, publicBaseUrl) => {
+  const { upload, extractFilesMeta, fileFieldsAppendSrc } = media(
+    dbDir,
+    publicBaseUrl,
+  );
+  const uploadMiddleware = upload.any();
+
+  const router = express.Router();
+
+  router.get("/", async (req, res) => {
+    const pagination =
+      req.query && req.query.pagination ? JSON.parse(req.query.pagination) : {};
+    const sort = req.query && req.query.sort ? JSON.parse(req.query.sort) : {};
+
+    const _start = (pagination.page - 1) * pagination.perPage || 0;
+    const _end = pagination.page * pagination.perPage || 25;
+    const _order = sort.order || "ASC";
+    const _sort = sort.field || "id";
+
+    const mediaPath = `${dbDir}/content/media`;
+    const filesInfo = await utils.readCollectionList(mediaPath);
+    const items = slice(orderBy(filesInfo, [_sort], [_order]), _start, _end);
+
+    res.setHeader("Access-Control-Expose-Headers", "X-Total-Count");
+    res.setHeader("X-Total-Count", items.length);
+    res.send({
+      data: items,
+      total: filesInfo.length,
+    });
+  });
+
+  router.get("/:itemId", async (req, res) => {
+    const { itemId } = req.params;
+
+    const mediaPath = `${dbDir}/content/media/${itemId}.json`;
+    const [data] = fileFieldsAppendSrc([await utils.read(mediaPath)]);
+    res.send({ data });
+  });
+
+  router.post("/", uploadMiddleware, async (req, res) => {
+    const mediaPath = `${dbDir}/content/media`;
+    const filesInfo = await utils.readCollectionList(mediaPath);
+    const newId = utils.getNewIdFromDatabaseItems(filesInfo);
+    const newMedia = {
+      ...extractFilesMeta(req.files).files,
```

Comment:
could you confirm 1 image file has exactly 1 json file with meta? "files" prop is plural and it just confuses me